regex-assembly/include/932130.ra

```diff
@@ -1,16 +1,16 @@
 ##! Please refer to the documentation at
 ##! https://coreruleset.org/docs/development/regex_assembly/.
 
-\$\(.*\)
-\$\{.*\}
-<\(.*\)
->\(.*\)
+\$\([^()]*\)
+\$\{[^{}]*\}
+<\([^()]*\)
+>\([^()]*\)
 ##! arithmetic expansion
-\$\(\(.*\)\)
-##! arithmetic expansion (deprecated in bash, but exists in other shells, like zsh) 
-\$\[.*\]
+\$\(\([^()]*\)\)
+##! arithmetic expansion (deprecated in bash, but exists in other shells, like zsh)
+\$\[[^\[\]]*\]
 
 ##! Find wordlist bypasses using [ ] glob characters, like: /e[t]c
 ##! Require a / in front to prevent false positives like [text in brackets]
-/\w*\[!.+\]
-/\w*\[.+\]
+/\w*\[![^\]\/]+?\]
+/\w*\[[^\]\/]+?\]
```

Comment:
1. No need to escape the forward slash.
2. Excluding `/` can cause increased backtracking on negative matches, e.g., `/[use about 3/4 of the liquid]`. The impact is probably neglectable.
3. I think that `+?` will actually lead to worse performance than simply using `+`. `+` will consume all tokens until the first `/` or `]` (if the token is `/`, the engine will backtrack). `+?` will consume one token at a time until it finds either `/` or `]`. It seems to me that the average performance of `+` would be better here, while worst case performance can be bad in both cases.